mysql-test/suite/encryption/t/percona_redo_default_key.test

```diff
@@ -0,0 +1,158 @@
+# it was rotated to already existing percona_redo-UUID:1 key (after SERVER_UUID became available).
+
+--remove_file $MYSQLD_DATADIR1/ib_logfile0
+--remove_file $MYSQLD_DATADIR1/ib_logfile1
+
+--let $restart_parameters = "restart: --loose-skip-log-bin --skip-log-slave-updates --datadir=$MYSQLD_DATADIR1 $KEYRING_DATA $KEYRING_PLUGIN_OPT $KEYRING_PLUGIN_EARLY_LOAD --log-error=$test_error_log --innodb_redo_log_encrypt=KEYRING_KEY"
+--replace_result $MYSQLD_DATADIR1 MYSQLD_DATADIR1 $MYSQL_TMP_DIR MYSQL_TMP_DIR $KEYRING_PLUGIN_OPT KEYRING_PLUGIN_OPT $KEYRING_PLUGIN_EARLY_LOAD KEYRING_PLUGIN_EARLY_LOAD
+--source include/start_mysqld.inc
+
+--source include/shutdown_mysqld.inc
+
+# Check that default percona_redo key exists only in version 0
+
+--let $percona_redo_default_key_name_v0=percona_redo:0
+
+--let SEARCH_PATTERN=$percona_redo_default_key_name_v0
+--let ABORT_ON=NOT_FOUND
+--let SEARCH_FILE=$KEYRING_FILE_PATH
+--source include/search_pattern_in_file.inc
+
+--let $percona_redo_default_key_name_v1=percona_redo:1
+
+--let SEARCH_PATTERN=$percona_redo_default_key_name_v1
+--let ABORT_ON=FOUND
+--source include/search_pattern_in_file.inc
+
+# Check that percona_redo-UUID key exists only in version 1
+
+--let $percona_key_name_v1=percona_redo-$server_uuid:1
+
+--let SEARCH_PATTERN=$percona_key_name_v1
+--let ABORT_ON=NOT_FOUND
+--source include/search_pattern_in_file.inc
+
+--let $percona_key_name_v2=percona_redo-$server_uuid:2
+
+--let SEARCH_PATTERN=$percona_key_name_v2
+--let ABORT_ON=FOUND
+--source include/search_pattern_in_file.inc
+
+# First we rotate percona_redo-UUID key to version 2. After that we shutdown the server and remove redo log files.
+# Next we start the server with activated code assertion - to check that when server UUID becomes available,
+# the percona_redo-UUID:2 will be used instead of percona_redo default key.
+
+--let $restart_parameters = "restart: --loose-skip-log-bin --skip-log-slave-updates --datadir=$MYSQLD_DATADIR1 $KEYRING_DATA $KEYRING_PLUGIN_OPT $KEYRING_PLUGIN_EARLY_LOAD --log-error=$test_error_log --innodb_redo_log_encrypt=KEYRING_KEY"
+--replace_result $MYSQLD_DATADIR1 MYSQLD_DATADIR1 $MYSQL_TMP_DIR MYSQL_TMP_DIR $KEYRING_PLUGIN_OPT KEYRING_PLUGIN_OPT $KEYRING_PLUGIN_EARLY_LOAD KEYRING_PLUGIN_EARLY_LOAD
+--source include/start_mysqld.inc
+
+--let $assert_text= rotate percona_redo-UUID key to version 2
+--let $assert_cond= "[SELECT rotate_system_key\\(\\"percona_redo\\"\\) = 1]" = 1
+--source include/assert.inc
+
+--source include/shutdown_mysqld.inc
+
+--remove_file $MYSQLD_DATADIR1/ib_logfile0
+--remove_file $MYSQLD_DATADIR1/ib_logfile1
+
+--let $restart_parameters = "restart: --loose-skip-log-bin --skip-log-slave-updates --datadir=$MYSQLD_DATADIR1 $KEYRING_DATA $KEYRING_PLUGIN_OPT $KEYRING_PLUGIN_EARLY_LOAD --log-error=$test_error_log --innodb_redo_log_encrypt=KEYRING_KEY --debug=+d,assert_default_to_ver2_rotation"
```

Comment:
isn't the startup supposed to crash/abort?

Try with --debug=d,assert..
if it is supposed to crash, you my have to use --exec

Reply:
actually it shouldn't. This just activates the assert that default key was rotated to ver.2 of percona_redo-UUID key.